### ISSUE TITLE
{{file::uuid}} insert tag gets converted to uuid|urlattr

### ISSUE THREAD

**Issue by fritzmg** [contao/contao#3430]
**Affected version(s)**

Contao `4.9.20`, `4.9.x-dev`

**Description**

If you use a regular `text` content element and insert a link to a file via the file picker in the TinyMCE, e.g.

```html
<p>Lorem <a href="{{file::c7869fac-ede7-11eb-9279-d8bbc1145c99}}">ipsum</a> dolor.</p>
```

then the content will be converted to

```html
<p>Lorem <a href="c7869fac-ede7-11eb-9279-d8bbc1145c99|urlattr">ipsum</a> dolor.</p>
```

after saving, which of course breaks the file reference.

/cc @ausi

**Comment by frontendschlampe** [comment]
I can confirm the issue

**Comment by m-vo** [comment]
for reference:

https://github.com/contao/contao/blob/bdc6e30cf50421bf3ec98d67b0b21ddd5ffa950d/core-bundle/src/Resources/contao/classes/DataContainer.php#L399-L403

https://github.com/contao/contao/blob/bdc6e30cf50421bf3ec98d67b0b21ddd5ffa950d/core-bundle/src/Resources/contao/library/Contao/StringUtil.php#L599-L626

**Comment by ausi** [comment]
Changing https://github.com/contao/contao/blob/0ba1acd7d93da55598a02fa6963dacf635bdad87/core-bundle/src/Resources/contao/library/Contao/StringUtil.php#L808 to:

```php
$paths = preg_split('/((src|href)="([^"]*){{file::([^"}|]+)[^"}]*}}")/i', $data, -1, PREG_SPLIT_DELIM_CAPTURE);
```

should fix the issue I think.

**Comment by fritzmg** [comment]
@ausi yes that works :). Will you open the PR so that the credit goes to you? ;)

**Comment by ausi** [comment]
Ja, ich will 😅

**Comment by ausi** [comment]
Closed in favor of #3455
